tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/ServiceRecipientsSectionControllerTests.cs

```diff
@@ -0,0 +1,181 @@
+            expected.ServiceRecipients = expectedList;
+
+            var response = await context.Controller.GetAllAsync(orderId);
+            response.Should().BeEquivalentTo(new ActionResult<ServiceRecipientsModel>(expected));
+        }
+        
+        [Test]
+        public async Task GetAllAsync_MultipleServiceRecipientsMatch_ReturnsAllTheOrdersServicesRecipients()
+        {
+            var context = ServiceRecipientsTestContext.Setup();
+
+            const string orderId = "C0000014-01";
+
+            context.Order = OrderBuilder.Create().WithOrderId(orderId).WithOrganisationId(context.PrimaryOrganisationId).Build();
+
+            var serviceRecipients = new List<(ServiceRecipient serviceRecipient, ServiceRecipientModel expectedModel)>
+            {
+                CreateServiceRecipientData("ODS1", orderId),
+                CreateServiceRecipientData("ODS2", orderId),
+                CreateServiceRecipientData("ODS3", orderId)
+            };
+
+            context.ServiceRecipients = serviceRecipients.Select(x => x.serviceRecipient).ToList();
+            var expected = new ServiceRecipientsModel();
+
+            var expectedList = serviceRecipients.Select(x => x.expectedModel);
+
+            expected.ServiceRecipients = expectedList;
+
+            var response = await context.Controller.GetAllAsync(orderId);
+            response.Should().BeEquivalentTo(new ActionResult<ServiceRecipientsModel>(expected));
+        }
+
+        private static (ServiceRecipient serviceRecipient, ServiceRecipientModel expectedModel)
+            CreateServiceRecipientData(string odsCode, string orderId)
+        {
+            var serviceRecipient = ServiceRecipientBuilder
+                .Create()
+                .WithOdsCode(odsCode)
+                .WithOrderId(orderId)
+                .Build();
+
+            return (serviceRecipient,
+                new ServiceRecipientModel { OdsCode = serviceRecipient.OdsCode, Name = serviceRecipient.Name });
+        }
+
+        private sealed class ServiceRecipientsTestContext
+        {
+            private ServiceRecipientsTestContext()
+            {
+                PrimaryOrganisationId = Guid.NewGuid();
+                Order = new Order { OrganisationId = PrimaryOrganisationId };
+
+                OrderRepositoryMock = new Mock<IOrderRepository>();
+                OrderRepositoryMock.Setup(x => x.GetOrderByIdAsync(It.IsAny<string>())).ReturnsAsync(() => Order);
+
+                ServiceRecipientRepositoryMock = new Mock<IServiceRecipientRepository>();
+                ServiceRecipients = new List<ServiceRecipient>();
+                ServiceRecipientRepositoryMock.Setup(x => x.ListServiceRecipientsByOrderId(It.IsAny<string>()))
+                    .ReturnsAsync(() => ServiceRecipients);
+
+                ClaimsPrincipal = new ClaimsPrincipal(new ClaimsIdentity(new[]
+                {
+                    new Claim("Ordering", "Manage"),

```

Comment:
@CallumBradburyBJSS Did you end up creating a builder for this common section of code?

Reply:
I didn't, I was going to bring it up at the code meeting this afternoon and get thoughts